lib/semantics/resolve-names.cc

```diff
@@ -2799,15 +2799,32 @@ bool SubprogramVisitor::BeginMpSubprogram(const parser::Name &name) {
   return true;
 }
 
+// A subprogram declared with SUBROUTINE or function
+// If the subprogram is not in interface block, check if its declared as separate 
+// module procedures
 bool SubprogramVisitor::BeginSubprogram(
     const parser::Name &name, Symbol::Flag subpFlag, bool hasModulePrefix) {
+  bool isSeparateModuleProc = false;
+  // Check if Subprogram has Module Prefix and Subprogram is not inside Interface Block 
   if (hasModulePrefix && !inInterfaceBlock()) {
-    Say(name, "'%s' was not declared a separate module procedure"_err_en_US);
-    return false;
+    // Check if the Subprogram is declared as separate module procedures
+    auto *symbol{FindSymbol(name)};
+    if (symbol && symbol->has<SubprogramNameDetails>()) {
+      symbol = FindSymbol(currScope().parent(), name);
+    }
+    if (symbol)
+      isSeparateModuleProc = symbol->IsSeparateModuleProc();
+
+    // Issue error: Subprogram is not declared as separate module procedure
```

Comment:
This comment and some of the other just repeat what the code is doing and so is unnecessary. Just use inline comments to clarify the code (or make the code clearer so they aren't necessary).

Reply:
Sure. i would so..